src/lib/Scenes/Artwork/__tests__/Artwork-tests.tsx

```diff
@@ -101,4 +112,72 @@ describe("Artwork", () => {
 
     expect(tree.root.findAllByType(ContextCard)).toHaveLength(1)
   })
+
+  describe("Live Auction States", () => {
+    it("has the correct state for a work that is in an auction that is currently live, for which I am registered", () => {
+      const tree = ReactTestRenderer.create(<TestRenderer />)
+
+      environment.mock.resolveMostRecentOperation(operation => {
+        const result = MockPayloadGenerator.generate(operation, {
+          Artwork() {
+            return merge({}, ArtworkFixture, ArtworkFromLiveAuctionRegistrationClosed, RegisteredBidder)
```

Comment:
I assume these fixtures have nested data (?), is that re-used by the mock generator for nested field selections?

Reply:
Yes.

Btw I'm not happy about using these fixtures here. I left them in for now because I got bogged down in trying to figure out exactly which properties on which types were needed to preserve the test behaviour, and without knowing the underlying auction logic it felt sinister to kinda reverse-engineer the artwork state like that. Might end up with something nonsensical.

Reply:
Gotcha. Good to know it works, but I agree about not leaving them in long-term.

I think we should get rid of fixtures due to complexity of building them _and_ maintaining them over time and code-changes/upgrades. What are your reasons?

Reply:
I'm not too worried about maintaining them over time. Honestly on the BNMO test suite I found that the big artwork/order fixtures saved time when writing new tests and were not much of a problem to maintain. (maybe you disagree having gone through MPv2?)  Most of my problems with fully-fleshed-out fixtures has been making sure IDs + their relationships are consistent and meaningful, which I'm not sure this mocking framework helps with.

The reason I don't want to keep them in here is I want to find a lighter-weight way of composing well-named 'states' of types. e.g. 'artwork which is in an auction that has already begun but the registration is still open and I have not registered for it yet' is not gonna be fun to deal with if it's defined anonymously as overridden type resolvers in every single test. There are just too many fields in play.

Reply:
> maybe you disagree having gone through MPv2?

Yeah, it’s been a major time-sink.